client/signup/steps/plans/index.jsx

```diff
@@ -146,6 +146,45 @@ export class PlansStep extends Component {
 			);
 		}
 
+		if ( isMobile() && 'onboarding' === this.props.flowName ) {
+			<ProvideExperimentData name="calypso_mobile_plans_page_with_billing">
```

Comment:
This is missing a `return` statement so this component won't be rendered. I think we should evaluate the return value to decide if this component should be rendered. So, maybe something like (I've not tested):
```
const billingWordExperimentComponent = (
       <ProvideExperimentData name="calypso_mobile_plans_page_with_billing">
       { ( isLoading, experimentAssignment ) => {
             if ( isLoading ) {
					return <PulsingDot active />;
			}

           const isControl = experimentAssignment?.variationName === null;
           if ( isControl ) {
                return null;
          }
			return (
				<div>
					{ errorDisplay }
					  .....
					/ >
                  </div>
             );
                  
       } }
        .... 
      </ProvideExperimentData>;

 if ( billingWordExperimentComponent !== null ) {
    return billingWordExperimentComponent;
}
```

Another method is to `dangerouslyGetExperimentAssignment()` ([source](dangerouslyGetExperimentAssignment)) and show the billing word experiment only if already assigned to `treatment`. This should work for new users because we prefetch the assignment in signup/controller.js, but will exclude existing users which I think should be okay.

```
const isTreatmentBillingWordExp = dangerouslyGetExperimentAssignment( 'calypso_mobile_plans_page_with_billing' );
if ( isTreatmentBillingWordExp && isMobile() && 'onboarding' === this.props.flowName ) {

 return <ProvideExperimentData name="calypso_mobile_plans_page_with_billing"
   ....
 >
}
```

Reply:
Good points, @niranjan-uma-shankar ! 

As a side note, there is also a third (recommended) approach, but it requires `PlansStep` to be a functional component or `plansFeaturesList` render method to be extracted to such component. Instead of having nested `<ProvideExperimentData />` helpers, calling `useExperiment` hook multiple times should get the experiment assignments and loading states while preventing code duplication.